uber/templates/macros.html

```diff
@@ -117,4 +117,113 @@
+    {% if not is_prereg_dealer %}
+        <script src="../static/deps/selectToAutocomplete/jquery.select-to-autocomplete.js"></script>
+        <script type="text/javascript">{% include "region_opts.html" %}</script>
+    {% endif %}
+    <script type="text/javascript">
+        var {{ name_prefix }}regionChange = function() {
+            {{ 'setInternational();' if not group else '' }}
+
+            var regionopts = $("[name='{{ name_prefix }}region']");
+            var whichCountry = $.field("{{ name_prefix }}country").find(':selected').text();
+            switch(whichCountry) {
+                case 'United States':
+                    regionopts.replaceWith('<select name="{{ name_prefix }}region" class="form-control"></select>');
+                    $("[name='{{ name_prefix }}region']").html(regionsforUS);
+                    break;
+                case 'Canada':
+                    regionopts.replaceWith('<select name="{{ name_prefix }}region" class="form-control"></select>');
+                    $("[name='{{ name_prefix }}region']").html(regionsforCanada);
+                    break;
+            }
+            var optionToSelect = $("select[name='{{ name_prefix }}region'] option[value='{{ group_or_attendee.region }}']");
+            if (optionToSelect.length) {
+                optionToSelect.prop('selected', true)
+            }
+        };
+
+        $(function() {
+            {{ 'setInternational();' if not group else '' }}
+            $.field("{{ name_prefix }}country").val('{{ group_or_attendee.country }}');
+            $.field("{{ name_prefix }}country").selectToAutocomplete();
+            $.field("{{ name_prefix }}country").change(function() {
+                $("[name='{{ name_prefix }}region']").replaceWith('<input type="text" name="{{ name_prefix }}region" value="{{ group_or_attendee.region }}" class="form-control" placeholder="State/Province">');
+                {{ name_prefix }}regionChange();
+            });
+
+            $.field("{{ name_prefix }}region").val('{{ group_or_attendee.region }}');
+            {{ name_prefix }}regionChange();
+        });
+    </script>
+
+    {% if is_prereg_dealer %}
+        <script type="text/javascript">
+        var copyAddress = function () {
+            if ($("#copy_address").prop('checked')) {
+                $('.address_details').hide();
+            } else if ($("#copy_address").length) {
+                $('.address_details').show();
+            }
+        };
+        $(function () {
+            if ($('#copy_address').length) {
+                copyAddress();
+                $('#copy_address').change(copyAddress);
+            }
+        })
+        </script>
+        <div class="form-group">
+            <div class="col-sm-6 col-sm-offset-3">
+                <label for="copy_address" class="checkbox-label">
+                    <input type="checkbox" name="copy_address" id="copy_address" value="1" {% if copy_address %}checked="checked"{% endif %} />
```

Comment:
Similarly to my comment about including javascript in a macro, we should probably avoid including hardcoded `id`s in macros.

If this macro were used twice on the same page, you'd end up with two `id="copy_address"` elements. Which is invalid HTML, and will cause problems when selecting the element by id (`"#copy_address"`).